examples/variational_autoencoder_deconv.py

```diff
@@ -48,23 +57,35 @@ def sampling(args):
 
 # we instantiate these layers separately so as to reuse them later
 decoder_h = Dense(intermediate_dim, activation='relu')
-decoder_f = Dense(nb_filters*img_rows*img_cols, activation='relu')
-decoder_c = Reshape((nb_filters, img_rows, img_cols))
-decoder_mean = Deconvolution2D(img_chns, nb_conv, nb_conv,
-                               (batch_size, img_chns, img_rows, img_cols),
-                               border_mode='same')
+decoder_f = Dense(nb_filters*14*14, activation='relu')
+decoder_c_1 = Reshape((nb_filters, 14, 14))
+decoder_c_2 = Deconvolution2D(nb_filters, nb_conv, nb_conv,
+                              (batch_size, nb_filters, 14, 14),
+                              border_mode='same',
+                              subsample=(1, 1))
+decoder_c_3 = Deconvolution2D(nb_filters, nb_conv, nb_conv,
+                              (batch_size, nb_filters, 14, 14),
+                              border_mode='same',
+                              subsample=(1, 1))
+decoder_mean = Deconvolution2D(nb_filters, 2, 2,
+                               (batch_size, nb_filters, 29, 29),
+                               border_mode='valid',
+                               subsample=(2, 2))
+decoder_mean_c = Convolution2D(img_chns, 2, 2, border_mode='valid', activation='relu')
 
 h_decoded = decoder_h(z)
 f_decoded = decoder_f(h_decoded)
-c_decoded = decoder_c(f_decoded)
-x_decoded_mean = decoder_mean(c_decoded)
-
+c_1_decoded = decoder_c_1(f_decoded)
+c_2_decoded = decoder_c_2(c_1_decoded)
+c_3_decoded = decoder_c_3(c_2_decoded)
+x_decoded_mean = decoder_mean(c_3_decoded)
+x_decoded_mean = decoder_mean_c(x_decoded_mean)
 
 def vae_loss(x, x_decoded_mean):
     # NOTE: binary_crossentropy expects a batch_size by dim for x and x_decoded_mean, so we MUST flatten these!
     x = K.flatten(x)
     x_decoded_mean = K.flatten(x_decoded_mean)
-    xent_loss = objectives.binary_crossentropy(x, x_decoded_mean)
+    xent_loss = 28 * 28 * objectives.binary_crossentropy(x, x_decoded_mean)
```

Comment:
Right, it's better to use the same settings as in the paper and to remove this arbitrary importance weight. From the paper:
"All parameters, both variational and generative, were initialized by random sampling from N (0, 0.01)".